{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Acá generamos el dataframe de base en el que se basará la red.\n",
    "\n",
    "Descarga de los datos: https://www.ebi.ac.uk/gwas/docs/file-downloads, \"All associations v1.0.2 - with added ontology\".\n",
    "\n",
    "Me quedo sólo con algunas columnas del dataset de asociaciones, les pongo nombres amigables y\n",
    "les asigno una categoría fenotípica a cada carácter (éstas vienen de un segundo archivo).\n",
    "\n",
    "Escribo el resultado a un nuevo dataset para basar todo el trabajo en ese."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "%run imports.py"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Carga y limpieza del dataset"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Asociaciones variante-fenotipo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Columnas disponibles:  DATE ADDED TO CATALOG | PUBMEDID | FIRST AUTHOR | DATE | JOURNAL | LINK | STUDY | DISEASE/TRAIT | INITIAL SAMPLE SIZE | REPLICATION SAMPLE SIZE | REGION | CHR_ID | CHR_POS | REPORTED GENE(S) | MAPPED_GENE | UPSTREAM_GENE_ID | DOWNSTREAM_GENE_ID | SNP_GENE_IDS | UPSTREAM_GENE_DISTANCE | DOWNSTREAM_GENE_DISTANCE | STRONGEST SNP-RISK ALLELE | SNPS | MERGED | SNP_ID_CURRENT | CONTEXT | INTERGENIC | RISK ALLELE FREQUENCY | P-VALUE | PVALUE_MLOG | P-VALUE (TEXT) | OR or BETA | 95% CI (TEXT) | PLATFORM [SNPS PASSING QC] | CNV | MAPPED_TRAIT | MAPPED_TRAIT_URI | STUDY ACCESSION | GENOTYPING TECHNOLOGY\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fenotipo</th>\n",
       "      <th>crom</th>\n",
       "      <th>pos</th>\n",
       "      <th>genes_reportados</th>\n",
       "      <th>genes_mapeados</th>\n",
       "      <th>alelo_riesgo</th>\n",
       "      <th>contexto</th>\n",
       "      <th>es_intergenico</th>\n",
       "      <th>frec_alelo_riesgo</th>\n",
       "      <th>p_value</th>\n",
       "      <th>OR_or_beta</th>\n",
       "      <th>conf_int_95</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Crohn's disease</td>\n",
       "      <td>10</td>\n",
       "      <td>6060049</td>\n",
       "      <td>IL2RA</td>\n",
       "      <td>IL2RA</td>\n",
       "      <td>rs12722489-C</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.852</td>\n",
       "      <td>3E-9</td>\n",
       "      <td>1.11</td>\n",
       "      <td>[1.05-1.16]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Crohn's disease</td>\n",
       "      <td>16</td>\n",
       "      <td>28479196</td>\n",
       "      <td>IL27, SH2B1, EIF3C, LAT, CD19</td>\n",
       "      <td>CLN3, AC138894.1</td>\n",
       "      <td>rs151181-G</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.386</td>\n",
       "      <td>2E-11</td>\n",
       "      <td>1.07</td>\n",
       "      <td>[1.03-1.12]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Crohn's disease</td>\n",
       "      <td>1</td>\n",
       "      <td>67240275</td>\n",
       "      <td>IL23R</td>\n",
       "      <td>IL23R</td>\n",
       "      <td>rs11209026-G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.932</td>\n",
       "      <td>1E-64</td>\n",
       "      <td>2.66</td>\n",
       "      <td>[2.36-3.00]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Crohn's disease</td>\n",
       "      <td>2</td>\n",
       "      <td>102437989</td>\n",
       "      <td>IL12RL2, IL18R1, IL1RL1, IL18RAP</td>\n",
       "      <td>IL18RAP</td>\n",
       "      <td>rs2058660-G</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.231</td>\n",
       "      <td>2E-12</td>\n",
       "      <td>1.19</td>\n",
       "      <td>[1.14-1.26]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Crohn's disease</td>\n",
       "      <td>5</td>\n",
       "      <td>159360377</td>\n",
       "      <td>IL12B</td>\n",
       "      <td>AC008691.1</td>\n",
       "      <td>rs6556412-A</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.332</td>\n",
       "      <td>5E-14</td>\n",
       "      <td>1.18</td>\n",
       "      <td>[1.13-1.24]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185719</th>\n",
       "      <td>Lung adenocarcinoma</td>\n",
       "      <td>6</td>\n",
       "      <td>31652743</td>\n",
       "      <td>APOM, BAT3</td>\n",
       "      <td>APOM, APOM, APOM, APOM, APOM, APOM, APOM</td>\n",
       "      <td>rs3117582-C</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.10</td>\n",
       "      <td>5E-12</td>\n",
       "      <td>1.22</td>\n",
       "      <td>[1.15-1.29]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185720</th>\n",
       "      <td>Lung adenocarcinoma</td>\n",
       "      <td>15</td>\n",
       "      <td>78601997</td>\n",
       "      <td>CHRNA5, CHRNA3</td>\n",
       "      <td>CHRNA3</td>\n",
       "      <td>rs1051730-T</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.35</td>\n",
       "      <td>2E-51</td>\n",
       "      <td>1.31</td>\n",
       "      <td>[1.27-1.36]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185721</th>\n",
       "      <td>Lung adenocarcinoma</td>\n",
       "      <td>5</td>\n",
       "      <td>1342599</td>\n",
       "      <td>CLPTM1L</td>\n",
       "      <td>CLPTM1L</td>\n",
       "      <td>rs31489-C</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.59</td>\n",
       "      <td>2E-10</td>\n",
       "      <td>1.12</td>\n",
       "      <td>[1.09-1.16]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185722</th>\n",
       "      <td>Lung adenocarcinoma</td>\n",
       "      <td>6</td>\n",
       "      <td>28808340</td>\n",
       "      <td>TRNAA-UGC</td>\n",
       "      <td>NOP56P1 - AL662890.1</td>\n",
       "      <td>rs4324798-A</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.09</td>\n",
       "      <td>2E-8</td>\n",
       "      <td>1.16</td>\n",
       "      <td>[1.09-1.24]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185723</th>\n",
       "      <td>Lung adenocarcinoma</td>\n",
       "      <td>5</td>\n",
       "      <td>1286401</td>\n",
       "      <td>TERT</td>\n",
       "      <td>TERT</td>\n",
       "      <td>rs2736100-G</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.50</td>\n",
       "      <td>2E-10</td>\n",
       "      <td>1.12</td>\n",
       "      <td>[1.08-1.16]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>185724 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   fenotipo crom        pos                  genes_reportados  \\\n",
       "0           Crohn's disease   10    6060049                             IL2RA   \n",
       "1           Crohn's disease   16   28479196     IL27, SH2B1, EIF3C, LAT, CD19   \n",
       "2           Crohn's disease    1   67240275                             IL23R   \n",
       "3           Crohn's disease    2  102437989  IL12RL2, IL18R1, IL1RL1, IL18RAP   \n",
       "4           Crohn's disease    5  159360377                             IL12B   \n",
       "...                     ...  ...        ...                               ...   \n",
       "185719  Lung adenocarcinoma    6   31652743                        APOM, BAT3   \n",
       "185720  Lung adenocarcinoma   15   78601997                    CHRNA5, CHRNA3   \n",
       "185721  Lung adenocarcinoma    5    1342599                           CLPTM1L   \n",
       "185722  Lung adenocarcinoma    6   28808340                         TRNAA-UGC   \n",
       "185723  Lung adenocarcinoma    5    1286401                              TERT   \n",
       "\n",
       "                                  genes_mapeados  alelo_riesgo  \\\n",
       "0                                          IL2RA  rs12722489-C   \n",
       "1                               CLN3, AC138894.1    rs151181-G   \n",
       "2                                          IL23R  rs11209026-G   \n",
       "3                                        IL18RAP   rs2058660-G   \n",
       "4                                     AC008691.1   rs6556412-A   \n",
       "...                                          ...           ...   \n",
       "185719  APOM, APOM, APOM, APOM, APOM, APOM, APOM   rs3117582-C   \n",
       "185720                                    CHRNA3   rs1051730-T   \n",
       "185721                                   CLPTM1L     rs31489-C   \n",
       "185722                      NOP56P1 - AL662890.1   rs4324798-A   \n",
       "185723                                      TERT   rs2736100-G   \n",
       "\n",
       "                   contexto  es_intergenico frec_alelo_riesgo p_value  \\\n",
       "0            intron_variant             0.0             0.852    3E-9   \n",
       "1            intron_variant             0.0             0.386   2E-11   \n",
       "2          missense_variant             0.0             0.932   1E-64   \n",
       "3            intron_variant             0.0             0.231   2E-12   \n",
       "4            intron_variant             0.0             0.332   5E-14   \n",
       "...                     ...             ...               ...     ...   \n",
       "185719       intron_variant             0.0              0.10   5E-12   \n",
       "185720   synonymous_variant             0.0              0.35   2E-51   \n",
       "185721       intron_variant             0.0              0.59   2E-10   \n",
       "185722   intergenic_variant             1.0              0.09    2E-8   \n",
       "185723  3_prime_UTR_variant             0.0              0.50   2E-10   \n",
       "\n",
       "        OR_or_beta  conf_int_95  \n",
       "0             1.11  [1.05-1.16]  \n",
       "1             1.07  [1.03-1.12]  \n",
       "2             2.66  [2.36-3.00]  \n",
       "3             1.19  [1.14-1.26]  \n",
       "4             1.18  [1.13-1.24]  \n",
       "...            ...          ...  \n",
       "185719        1.22  [1.15-1.29]  \n",
       "185720        1.31  [1.27-1.36]  \n",
       "185721        1.12  [1.09-1.16]  \n",
       "185722        1.16  [1.09-1.24]  \n",
       "185723        1.12  [1.08-1.16]  \n",
       "\n",
       "[185724 rows x 12 columns]"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "path = f\"{data_dir}/gwas_catalog_v1.0.2-associations_e98_r2020-05-03.tsv.gz\"\n",
    "df_assoc_full = pd.read_table(path, low_memory=False)\n",
    "print(\"Columnas disponibles: \", \" | \".join(df_assoc_full.columns))\n",
    "\n",
    "columnas = {\n",
    "    \"DISEASE/TRAIT\": \"fenotipo\",\n",
    "    \"CHR_ID\": \"crom\",\n",
    "    \"CHR_POS\": \"pos\",\n",
    "    \"REPORTED GENE(S)\": \"genes_reportados\",\n",
    "    \"MAPPED_GENE\": \"genes_mapeados\",\n",
    "    \"STRONGEST SNP-RISK ALLELE\": \"alelo_riesgo\",\n",
    "    \"CONTEXT\": \"contexto\",\n",
    "    \"INTERGENIC\": \"es_intergenico\",\n",
    "    \"RISK ALLELE FREQUENCY\": \"frec_alelo_riesgo\",\n",
    "    \"P-VALUE\": \"p_value\",\n",
    "    \"OR or BETA\": \"OR_or_beta\",\n",
    "    \"95% CI (TEXT)\": \"conf_int_95\"\n",
    "}\n",
    "df_assoc = df_assoc_full[columnas].rename(columns=columnas)\n",
    "df_assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "from more_itertools import unique_everseen\n",
    "\n",
    "def eliminar_items_repetidos(items_str, sep):\n",
    "    if pd.isna(items_str):\n",
    "        return\n",
    "    \n",
    "    items_unicos = unique_everseen(items_str.split(sep))\n",
    "    return \" | \".join(item.strip() for item in items_unicos)\n",
    "\n",
    "# Hay valores como \"APOM, APOM, APOM\" en los genes, también \"GEN1; GEN1\". Limpiamos\n",
    "# y unificamos el separador como \" | \":\n",
    "\n",
    "for sep in [\",\", \";\"]:\n",
    "    df_assoc[\"genes_mapeados\"] = df_assoc.genes_mapeados.apply(eliminar_items_repetidos, sep=sep)\n",
    "    df_assoc[\"genes_reportados\"] = df_assoc.genes_reportados.apply(eliminar_items_repetidos, sep=sep)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'bar'"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d = {\"foo\": 2, \"bar\": 4, \"baz\": 1}\n",
    "# max(d, key=itemgetter(1))\n",
    "pd.Series(d).idxmax()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1314"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from more_itertools import flatten\n",
    "from collections import Counter\n",
    "\n",
    "genes_mapeados_por_alelo = df_assoc.genes_mapeados.dropna().apply(lambda genes_str: genes_str.split(\" | \"))\n",
    "genes_mapeados_frecuencia = pd.Series(flatten(genes_mapeados_por_alelo)).value_counts()\n",
    "genes_ordenados_por_frecuencia = genes_mapeados_frecuencia.index\n",
    "\n",
    "genes_mapeados_frecuencia[\"SARM1\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 185724/185724 [01:03<00:00, 2912.22it/s]\n"
     ]
    }
   ],
   "source": [
    "def elegir_un_gen(genes_str, sep):\n",
    "    \"\"\"Quita genes que se ven poco interesantes por razones variadas y de los restantes\n",
    "       elige el más frecuente del dataset. Si hay empates elige el primero.\"\"\"\n",
    "    if pd.isna(genes_str):\n",
    "        return\n",
    "   \n",
    "    genes = genes_str.split(sep)\n",
    "    genes_interesantes = [\n",
    "        gen for gen in genes if\n",
    "        \".\" not in gen and # probablemente ids no interesantes\n",
    "        \" - \" not in gen and # alelo intergénico reportado entre GEN1 - GEN2\n",
    "        \"-\" not in gen and # genes antisense tipo FOOBAR-AS1\n",
    "        \"orf\" not in gen # ORFs sin función identificada\n",
    "    ]\n",
    "    \n",
    "    if not genes_interesantes:\n",
    "        return\n",
    "    \n",
    "    frecuencias_por_gen = pd.Series({gen: genes_mapeados_frecuencia[gen]\n",
    "                                     for gen in genes_interesantes}, dtype=int)\n",
    "    return frecuencias_por_gen.idxmax()\n",
    "    \n",
    "\n",
    "df_assoc[\"gen_mapeado_elegido\"] = df_assoc.genes_mapeados.progress_apply(elegir_un_gen, sep=\" | \")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fenotipo</th>\n",
       "      <th>crom</th>\n",
       "      <th>pos</th>\n",
       "      <th>genes_reportados</th>\n",
       "      <th>genes_mapeados</th>\n",
       "      <th>alelo_riesgo</th>\n",
       "      <th>contexto</th>\n",
       "      <th>es_intergenico</th>\n",
       "      <th>frec_alelo_riesgo</th>\n",
       "      <th>p_value</th>\n",
       "      <th>OR_or_beta</th>\n",
       "      <th>conf_int_95</th>\n",
       "      <th>gen_mapeado_elegido</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Crohn's disease</td>\n",
       "      <td>10</td>\n",
       "      <td>6060049</td>\n",
       "      <td>IL2RA</td>\n",
       "      <td>IL2RA</td>\n",
       "      <td>rs12722489-C</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.852</td>\n",
       "      <td>3E-9</td>\n",
       "      <td>1.11</td>\n",
       "      <td>[1.05-1.16]</td>\n",
       "      <td>IL2RA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Crohn's disease</td>\n",
       "      <td>16</td>\n",
       "      <td>28479196</td>\n",
       "      <td>IL27 | SH2B1 | EIF3C | LAT | CD19</td>\n",
       "      <td>CLN3 | AC138894.1</td>\n",
       "      <td>rs151181-G</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.386</td>\n",
       "      <td>2E-11</td>\n",
       "      <td>1.07</td>\n",
       "      <td>[1.03-1.12]</td>\n",
       "      <td>CLN3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Crohn's disease</td>\n",
       "      <td>1</td>\n",
       "      <td>67240275</td>\n",
       "      <td>IL23R</td>\n",
       "      <td>IL23R</td>\n",
       "      <td>rs11209026-G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.932</td>\n",
       "      <td>1E-64</td>\n",
       "      <td>2.66</td>\n",
       "      <td>[2.36-3.00]</td>\n",
       "      <td>IL23R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Crohn's disease</td>\n",
       "      <td>2</td>\n",
       "      <td>102437989</td>\n",
       "      <td>IL12RL2 | IL18R1 | IL1RL1 | IL18RAP</td>\n",
       "      <td>IL18RAP</td>\n",
       "      <td>rs2058660-G</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.231</td>\n",
       "      <td>2E-12</td>\n",
       "      <td>1.19</td>\n",
       "      <td>[1.14-1.26]</td>\n",
       "      <td>IL18RAP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Crohn's disease</td>\n",
       "      <td>5</td>\n",
       "      <td>159360377</td>\n",
       "      <td>IL12B</td>\n",
       "      <td>AC008691.1</td>\n",
       "      <td>rs6556412-A</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.332</td>\n",
       "      <td>5E-14</td>\n",
       "      <td>1.18</td>\n",
       "      <td>[1.13-1.24]</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185719</th>\n",
       "      <td>Lung adenocarcinoma</td>\n",
       "      <td>6</td>\n",
       "      <td>31652743</td>\n",
       "      <td>APOM | BAT3</td>\n",
       "      <td>APOM | APOM</td>\n",
       "      <td>rs3117582-C</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.10</td>\n",
       "      <td>5E-12</td>\n",
       "      <td>1.22</td>\n",
       "      <td>[1.15-1.29]</td>\n",
       "      <td>APOM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185720</th>\n",
       "      <td>Lung adenocarcinoma</td>\n",
       "      <td>15</td>\n",
       "      <td>78601997</td>\n",
       "      <td>CHRNA5 | CHRNA3</td>\n",
       "      <td>CHRNA3</td>\n",
       "      <td>rs1051730-T</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.35</td>\n",
       "      <td>2E-51</td>\n",
       "      <td>1.31</td>\n",
       "      <td>[1.27-1.36]</td>\n",
       "      <td>CHRNA3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185721</th>\n",
       "      <td>Lung adenocarcinoma</td>\n",
       "      <td>5</td>\n",
       "      <td>1342599</td>\n",
       "      <td>CLPTM1L</td>\n",
       "      <td>CLPTM1L</td>\n",
       "      <td>rs31489-C</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.59</td>\n",
       "      <td>2E-10</td>\n",
       "      <td>1.12</td>\n",
       "      <td>[1.09-1.16]</td>\n",
       "      <td>CLPTM1L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185722</th>\n",
       "      <td>Lung adenocarcinoma</td>\n",
       "      <td>6</td>\n",
       "      <td>28808340</td>\n",
       "      <td>TRNAA-UGC</td>\n",
       "      <td>NOP56P1 - AL662890.1</td>\n",
       "      <td>rs4324798-A</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.09</td>\n",
       "      <td>2E-8</td>\n",
       "      <td>1.16</td>\n",
       "      <td>[1.09-1.24]</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185723</th>\n",
       "      <td>Lung adenocarcinoma</td>\n",
       "      <td>5</td>\n",
       "      <td>1286401</td>\n",
       "      <td>TERT</td>\n",
       "      <td>TERT</td>\n",
       "      <td>rs2736100-G</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.50</td>\n",
       "      <td>2E-10</td>\n",
       "      <td>1.12</td>\n",
       "      <td>[1.08-1.16]</td>\n",
       "      <td>TERT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>185724 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   fenotipo crom        pos  \\\n",
       "0           Crohn's disease   10    6060049   \n",
       "1           Crohn's disease   16   28479196   \n",
       "2           Crohn's disease    1   67240275   \n",
       "3           Crohn's disease    2  102437989   \n",
       "4           Crohn's disease    5  159360377   \n",
       "...                     ...  ...        ...   \n",
       "185719  Lung adenocarcinoma    6   31652743   \n",
       "185720  Lung adenocarcinoma   15   78601997   \n",
       "185721  Lung adenocarcinoma    5    1342599   \n",
       "185722  Lung adenocarcinoma    6   28808340   \n",
       "185723  Lung adenocarcinoma    5    1286401   \n",
       "\n",
       "                           genes_reportados        genes_mapeados  \\\n",
       "0                                     IL2RA                 IL2RA   \n",
       "1         IL27 | SH2B1 | EIF3C | LAT | CD19     CLN3 | AC138894.1   \n",
       "2                                     IL23R                 IL23R   \n",
       "3       IL12RL2 | IL18R1 | IL1RL1 | IL18RAP               IL18RAP   \n",
       "4                                     IL12B            AC008691.1   \n",
       "...                                     ...                   ...   \n",
       "185719                          APOM | BAT3           APOM | APOM   \n",
       "185720                      CHRNA5 | CHRNA3                CHRNA3   \n",
       "185721                              CLPTM1L               CLPTM1L   \n",
       "185722                            TRNAA-UGC  NOP56P1 - AL662890.1   \n",
       "185723                                 TERT                  TERT   \n",
       "\n",
       "        alelo_riesgo             contexto  es_intergenico frec_alelo_riesgo  \\\n",
       "0       rs12722489-C       intron_variant             0.0             0.852   \n",
       "1         rs151181-G       intron_variant             0.0             0.386   \n",
       "2       rs11209026-G     missense_variant             0.0             0.932   \n",
       "3        rs2058660-G       intron_variant             0.0             0.231   \n",
       "4        rs6556412-A       intron_variant             0.0             0.332   \n",
       "...              ...                  ...             ...               ...   \n",
       "185719   rs3117582-C       intron_variant             0.0              0.10   \n",
       "185720   rs1051730-T   synonymous_variant             0.0              0.35   \n",
       "185721     rs31489-C       intron_variant             0.0              0.59   \n",
       "185722   rs4324798-A   intergenic_variant             1.0              0.09   \n",
       "185723   rs2736100-G  3_prime_UTR_variant             0.0              0.50   \n",
       "\n",
       "       p_value  OR_or_beta  conf_int_95 gen_mapeado_elegido  \n",
       "0         3E-9        1.11  [1.05-1.16]               IL2RA  \n",
       "1        2E-11        1.07  [1.03-1.12]                CLN3  \n",
       "2        1E-64        2.66  [2.36-3.00]               IL23R  \n",
       "3        2E-12        1.19  [1.14-1.26]             IL18RAP  \n",
       "4        5E-14        1.18  [1.13-1.24]                None  \n",
       "...        ...         ...          ...                 ...  \n",
       "185719   5E-12        1.22  [1.15-1.29]                APOM  \n",
       "185720   2E-51        1.31  [1.27-1.36]              CHRNA3  \n",
       "185721   2E-10        1.12  [1.09-1.16]             CLPTM1L  \n",
       "185722    2E-8        1.16  [1.09-1.24]                None  \n",
       "185723   2E-10        1.12  [1.08-1.16]                TERT  \n",
       "\n",
       "[185724 rows x 13 columns]"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_assoc"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Categorías más generales de los fenotipos"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Disease trait</th>\n",
       "      <th>EFO term</th>\n",
       "      <th>EFO URI</th>\n",
       "      <th>Parent term</th>\n",
       "      <th>Parent URI</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Clubfoot</td>\n",
       "      <td>familial clubfoot with or without associated lower limb anomalies</td>\n",
       "      <td>http://www.orpha.net/ORDO/Orphanet_199315</td>\n",
       "      <td>Other disease</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0000408</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Alcohol dependence or heroin dependence or methamphetamine dependence</td>\n",
       "      <td>methamphetamine dependence</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0004701</td>\n",
       "      <td>Neurological disorder</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0000618</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Methamphetamine dependence</td>\n",
       "      <td>methamphetamine dependence</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0004701</td>\n",
       "      <td>Neurological disorder</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0000618</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Hypothyroidism</td>\n",
       "      <td>hypothyroidism</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0004705</td>\n",
       "      <td>Other disease</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0000408</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Age at menopause</td>\n",
       "      <td>age at menopause</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0004704</td>\n",
       "      <td>Other measurement</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0001444</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                           Disease trait  \\\n",
       "0                                                               Clubfoot   \n",
       "1  Alcohol dependence or heroin dependence or methamphetamine dependence   \n",
       "2                                             Methamphetamine dependence   \n",
       "3                                                         Hypothyroidism   \n",
       "4                                                       Age at menopause   \n",
       "\n",
       "                                                            EFO term  \\\n",
       "0  familial clubfoot with or without associated lower limb anomalies   \n",
       "1                                         methamphetamine dependence   \n",
       "2                                         methamphetamine dependence   \n",
       "3                                                     hypothyroidism   \n",
       "4                                                   age at menopause   \n",
       "\n",
       "                                     EFO URI            Parent term  \\\n",
       "0  http://www.orpha.net/ORDO/Orphanet_199315          Other disease   \n",
       "1       http://www.ebi.ac.uk/efo/EFO_0004701  Neurological disorder   \n",
       "2       http://www.ebi.ac.uk/efo/EFO_0004701  Neurological disorder   \n",
       "3       http://www.ebi.ac.uk/efo/EFO_0004705          Other disease   \n",
       "4       http://www.ebi.ac.uk/efo/EFO_0004704      Other measurement   \n",
       "\n",
       "                             Parent URI  \n",
       "0  http://www.ebi.ac.uk/efo/EFO_0000408  \n",
       "1  http://www.ebi.ac.uk/efo/EFO_0000618  \n",
       "2  http://www.ebi.ac.uk/efo/EFO_0000618  \n",
       "3  http://www.ebi.ac.uk/efo/EFO_0000408  \n",
       "4  http://www.ebi.ac.uk/efo/EFO_0001444  "
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "path = f\"{data_dir}/gwas_catalog_trait-mappings_r2020-05-17.tsv.gz\"\n",
    "trait_mappings = pd.read_table(path)\n",
    "trait_mappings.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>categoria_fenotipo</th>\n",
       "      <th>fenotipo</th>\n",
       "      <th>crom</th>\n",
       "      <th>pos</th>\n",
       "      <th>genes_reportados</th>\n",
       "      <th>genes_mapeados</th>\n",
       "      <th>alelo_riesgo</th>\n",
       "      <th>contexto</th>\n",
       "      <th>es_intergenico</th>\n",
       "      <th>frec_alelo_riesgo</th>\n",
       "      <th>p_value</th>\n",
       "      <th>OR_or_beta</th>\n",
       "      <th>conf_int_95</th>\n",
       "      <th>gen_mapeado_elegido</th>\n",
       "      <th>EFO_URI</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Digestive system disorder</td>\n",
       "      <td>Crohn's disease</td>\n",
       "      <td>10</td>\n",
       "      <td>6060049</td>\n",
       "      <td>IL2RA</td>\n",
       "      <td>IL2RA</td>\n",
       "      <td>rs12722489-C</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>False</td>\n",
       "      <td>0.852</td>\n",
       "      <td>3E-9</td>\n",
       "      <td>1.11</td>\n",
       "      <td>[1.05-1.16]</td>\n",
       "      <td>IL2RA</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0000384</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Digestive system disorder</td>\n",
       "      <td>Crohn's disease</td>\n",
       "      <td>16</td>\n",
       "      <td>28479196</td>\n",
       "      <td>IL27 | SH2B1 | EIF3C | LAT | CD19</td>\n",
       "      <td>CLN3 | AC138894.1</td>\n",
       "      <td>rs151181-G</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>False</td>\n",
       "      <td>0.386</td>\n",
       "      <td>2E-11</td>\n",
       "      <td>1.07</td>\n",
       "      <td>[1.03-1.12]</td>\n",
       "      <td>CLN3</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0000384</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Digestive system disorder</td>\n",
       "      <td>Crohn's disease</td>\n",
       "      <td>1</td>\n",
       "      <td>67240275</td>\n",
       "      <td>IL23R</td>\n",
       "      <td>IL23R</td>\n",
       "      <td>rs11209026-G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>False</td>\n",
       "      <td>0.932</td>\n",
       "      <td>1E-64</td>\n",
       "      <td>2.66</td>\n",
       "      <td>[2.36-3.00]</td>\n",
       "      <td>IL23R</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0000384</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Digestive system disorder</td>\n",
       "      <td>Crohn's disease</td>\n",
       "      <td>2</td>\n",
       "      <td>102437989</td>\n",
       "      <td>IL12RL2 | IL18R1 | IL1RL1 | IL18RAP</td>\n",
       "      <td>IL18RAP</td>\n",
       "      <td>rs2058660-G</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>False</td>\n",
       "      <td>0.231</td>\n",
       "      <td>2E-12</td>\n",
       "      <td>1.19</td>\n",
       "      <td>[1.14-1.26]</td>\n",
       "      <td>IL18RAP</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0000384</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Digestive system disorder</td>\n",
       "      <td>Crohn's disease</td>\n",
       "      <td>5</td>\n",
       "      <td>159360377</td>\n",
       "      <td>IL12B</td>\n",
       "      <td>AC008691.1</td>\n",
       "      <td>rs6556412-A</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>False</td>\n",
       "      <td>0.332</td>\n",
       "      <td>5E-14</td>\n",
       "      <td>1.18</td>\n",
       "      <td>[1.13-1.24]</td>\n",
       "      <td>None</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0000384</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185719</th>\n",
       "      <td>Cancer</td>\n",
       "      <td>Lung adenocarcinoma</td>\n",
       "      <td>6</td>\n",
       "      <td>31652743</td>\n",
       "      <td>APOM | BAT3</td>\n",
       "      <td>APOM | APOM</td>\n",
       "      <td>rs3117582-C</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>False</td>\n",
       "      <td>0.10</td>\n",
       "      <td>5E-12</td>\n",
       "      <td>1.22</td>\n",
       "      <td>[1.15-1.29]</td>\n",
       "      <td>APOM</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0000571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185720</th>\n",
       "      <td>Cancer</td>\n",
       "      <td>Lung adenocarcinoma</td>\n",
       "      <td>15</td>\n",
       "      <td>78601997</td>\n",
       "      <td>CHRNA5 | CHRNA3</td>\n",
       "      <td>CHRNA3</td>\n",
       "      <td>rs1051730-T</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>False</td>\n",
       "      <td>0.35</td>\n",
       "      <td>2E-51</td>\n",
       "      <td>1.31</td>\n",
       "      <td>[1.27-1.36]</td>\n",
       "      <td>CHRNA3</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0000571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185721</th>\n",
       "      <td>Cancer</td>\n",
       "      <td>Lung adenocarcinoma</td>\n",
       "      <td>5</td>\n",
       "      <td>1342599</td>\n",
       "      <td>CLPTM1L</td>\n",
       "      <td>CLPTM1L</td>\n",
       "      <td>rs31489-C</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>False</td>\n",
       "      <td>0.59</td>\n",
       "      <td>2E-10</td>\n",
       "      <td>1.12</td>\n",
       "      <td>[1.09-1.16]</td>\n",
       "      <td>CLPTM1L</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0000571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185722</th>\n",
       "      <td>Cancer</td>\n",
       "      <td>Lung adenocarcinoma</td>\n",
       "      <td>6</td>\n",
       "      <td>28808340</td>\n",
       "      <td>TRNAA-UGC</td>\n",
       "      <td>NOP56P1 - AL662890.1</td>\n",
       "      <td>rs4324798-A</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>True</td>\n",
       "      <td>0.09</td>\n",
       "      <td>2E-8</td>\n",
       "      <td>1.16</td>\n",
       "      <td>[1.09-1.24]</td>\n",
       "      <td>None</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0000571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185723</th>\n",
       "      <td>Cancer</td>\n",
       "      <td>Lung adenocarcinoma</td>\n",
       "      <td>5</td>\n",
       "      <td>1286401</td>\n",
       "      <td>TERT</td>\n",
       "      <td>TERT</td>\n",
       "      <td>rs2736100-G</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>False</td>\n",
       "      <td>0.50</td>\n",
       "      <td>2E-10</td>\n",
       "      <td>1.12</td>\n",
       "      <td>[1.08-1.16]</td>\n",
       "      <td>TERT</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0000571</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>185724 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               categoria_fenotipo             fenotipo crom        pos  \\\n",
       "0       Digestive system disorder      Crohn's disease   10    6060049   \n",
       "1       Digestive system disorder      Crohn's disease   16   28479196   \n",
       "2       Digestive system disorder      Crohn's disease    1   67240275   \n",
       "3       Digestive system disorder      Crohn's disease    2  102437989   \n",
       "4       Digestive system disorder      Crohn's disease    5  159360377   \n",
       "...                           ...                  ...  ...        ...   \n",
       "185719                     Cancer  Lung adenocarcinoma    6   31652743   \n",
       "185720                     Cancer  Lung adenocarcinoma   15   78601997   \n",
       "185721                     Cancer  Lung adenocarcinoma    5    1342599   \n",
       "185722                     Cancer  Lung adenocarcinoma    6   28808340   \n",
       "185723                     Cancer  Lung adenocarcinoma    5    1286401   \n",
       "\n",
       "                           genes_reportados        genes_mapeados  \\\n",
       "0                                     IL2RA                 IL2RA   \n",
       "1         IL27 | SH2B1 | EIF3C | LAT | CD19     CLN3 | AC138894.1   \n",
       "2                                     IL23R                 IL23R   \n",
       "3       IL12RL2 | IL18R1 | IL1RL1 | IL18RAP               IL18RAP   \n",
       "4                                     IL12B            AC008691.1   \n",
       "...                                     ...                   ...   \n",
       "185719                          APOM | BAT3           APOM | APOM   \n",
       "185720                      CHRNA5 | CHRNA3                CHRNA3   \n",
       "185721                              CLPTM1L               CLPTM1L   \n",
       "185722                            TRNAA-UGC  NOP56P1 - AL662890.1   \n",
       "185723                                 TERT                  TERT   \n",
       "\n",
       "        alelo_riesgo             contexto  es_intergenico frec_alelo_riesgo  \\\n",
       "0       rs12722489-C       intron_variant           False             0.852   \n",
       "1         rs151181-G       intron_variant           False             0.386   \n",
       "2       rs11209026-G     missense_variant           False             0.932   \n",
       "3        rs2058660-G       intron_variant           False             0.231   \n",
       "4        rs6556412-A       intron_variant           False             0.332   \n",
       "...              ...                  ...             ...               ...   \n",
       "185719   rs3117582-C       intron_variant           False              0.10   \n",
       "185720   rs1051730-T   synonymous_variant           False              0.35   \n",
       "185721     rs31489-C       intron_variant           False              0.59   \n",
       "185722   rs4324798-A   intergenic_variant            True              0.09   \n",
       "185723   rs2736100-G  3_prime_UTR_variant           False              0.50   \n",
       "\n",
       "       p_value  OR_or_beta  conf_int_95 gen_mapeado_elegido  \\\n",
       "0         3E-9        1.11  [1.05-1.16]               IL2RA   \n",
       "1        2E-11        1.07  [1.03-1.12]                CLN3   \n",
       "2        1E-64        2.66  [2.36-3.00]               IL23R   \n",
       "3        2E-12        1.19  [1.14-1.26]             IL18RAP   \n",
       "4        5E-14        1.18  [1.13-1.24]                None   \n",
       "...        ...         ...          ...                 ...   \n",
       "185719   5E-12        1.22  [1.15-1.29]                APOM   \n",
       "185720   2E-51        1.31  [1.27-1.36]              CHRNA3   \n",
       "185721   2E-10        1.12  [1.09-1.16]             CLPTM1L   \n",
       "185722    2E-8        1.16  [1.09-1.24]                None   \n",
       "185723   2E-10        1.12  [1.08-1.16]                TERT   \n",
       "\n",
       "                                     EFO_URI  \n",
       "0       http://www.ebi.ac.uk/efo/EFO_0000384  \n",
       "1       http://www.ebi.ac.uk/efo/EFO_0000384  \n",
       "2       http://www.ebi.ac.uk/efo/EFO_0000384  \n",
       "3       http://www.ebi.ac.uk/efo/EFO_0000384  \n",
       "4       http://www.ebi.ac.uk/efo/EFO_0000384  \n",
       "...                                      ...  \n",
       "185719  http://www.ebi.ac.uk/efo/EFO_0000571  \n",
       "185720  http://www.ebi.ac.uk/efo/EFO_0000571  \n",
       "185721  http://www.ebi.ac.uk/efo/EFO_0000571  \n",
       "185722  http://www.ebi.ac.uk/efo/EFO_0000571  \n",
       "185723  http://www.ebi.ac.uk/efo/EFO_0000571  \n",
       "\n",
       "[185724 rows x 15 columns]"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fenotipo_a_categoria = dict(zip(trait_mappings[\"Disease trait\"], trait_mappings[\"Parent term\"]))\n",
    "fenotipo_a_EFO_URI = dict(zip(trait_mappings[\"Disease trait\"], trait_mappings[\"EFO URI\"]))\n",
    "\n",
    "categorias = df_assoc.fenotipo.map(fenotipo_a_categoria)\n",
    "\n",
    "df_assoc[\"EFO_URI\"] = df_assoc.fenotipo.map(fenotipo_a_EFO_URI)\n",
    "df_assoc.insert(0, \"categoria_fenotipo\", categorias)\n",
    "\n",
    "df_assoc[\"es_intergenico\"] = df_assoc.es_intergenico.astype(bool)\n",
    "\n",
    "df_assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "results/GwasCat_associations.r2020-05-03.parsed.csv.gz\n"
     ]
    }
   ],
   "source": [
    "path = f\"{results_dir}/GwasCat_associations.r2020-05-03.parsed.csv.gz\"\n",
    "print(path)\n",
    "df_assoc.to_csv(path, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "gzip: stdout: Broken pipe\n",
      "categoria_fenotipo         fenotipo         crom  pos        genes_reportados                     genes_mapeados           alelo_riesgo  contexto                   es_intergenico  frec_alelo_riesgo  p_value  OR_or_beta  conf_int_95  gen_mapeado_elegido                   EFO_URI\n",
      "Digestive system disorder  Crohn's disease  10    6060049    IL2RA                                IL2RA                    rs12722489-C  intron_variant             False           0.852              3E-9     1.11        [1.05-1.16]  IL2RA                                 http://www.ebi.ac.uk/efo/EFO_0000384\n",
      "Digestive system disorder  Crohn's disease  16    28479196   IL27 | SH2B1 | EIF3C | LAT | CD19    CLN3 | AC138894.1        rs151181-G    intron_variant             False           0.386              2E-11    1.07        [1.03-1.12]  CLN3                                  http://www.ebi.ac.uk/efo/EFO_0000384\n",
      "Digestive system disorder  Crohn's disease  1     67240275   IL23R                                IL23R                    rs11209026-G  missense_variant           False           0.932              1E-64    2.66        [2.36-3.00]  IL23R                                 http://www.ebi.ac.uk/efo/EFO_0000384\n",
      "Digestive system disorder  Crohn's disease  2     102437989  IL12RL2 | IL18R1 | IL1RL1 | IL18RAP  IL18RAP                  rs2058660-G   intron_variant             False           0.231              2E-12    1.19        [1.14-1.26]  IL18RAP                               http://www.ebi.ac.uk/efo/EFO_0000384\n",
      "Digestive system disorder  Crohn's disease  5     159360377  IL12B                                AC008691.1               rs6556412-A   intron_variant             False           0.332              5E-14    1.18        [1.13-1.24]  http://www.ebi.ac.uk/efo/EFO_0000384\n",
      "Digestive system disorder  Crohn's disease  1     7819003    VAMP3                                PER3 | Z98884.1          rs2797685-A   intron_variant             False           0.19               7E-9     1.05        [1.01-1.10]  PER3                                  http://www.ebi.ac.uk/efo/EFO_0000384\n",
      "Digestive system disorder  Crohn's disease  1     206766559  IL10 | IL19                          Y_RNA - IL10             rs3024505-T   intergenic_variant         True            0.157              2E-14    1.12        [1.07-1.17]  http://www.ebi.ac.uk/efo/EFO_0000384\n",
      "Digestive system disorder  Crohn's disease  7     50264865   ZPBP | IKZF1 | FIGNL1                AC020743.2 - AC020743.3  rs1456896-T   regulatory_region_variant  True            0.69               1E-8     1.14        [1.09-1.20]  http://www.ebi.ac.uk/efo/EFO_0000384\n",
      "Digestive system disorder  Crohn's disease  21    44195140   ICOSLG                               GATD3A                   rs2838519-G   intron_variant             False           0.391              2E-14    1.18        [1.13-1.23]  GATD3A                                http://www.ebi.ac.uk/efo/EFO_0000384\n"
     ]
    }
   ],
   "source": [
    "! zcat $path | head | column -s\",\" -t"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
